Gauss pp

p3jawors opened the pull request:
Overhaul of path planners. We now have one master path planner that takes in a position and velocity profile. This allows up to separate the shape of the path we're taking, and the velocity throughout that path. Examples have been updated, and a new section for path_planner examples has been added.

p3jawors commented:
- Updated README to include description for new path planner
- Updated dynamics_adaptation to close [issue 90](https://github.com/abr/abr_control/issues/90)
- Renamed examples to close [issue 51](https://github.com/abr/abr_control/issues/51)

studywolf commented on abr_control/controllers/path_planners/orientation.py:
```diff
@@ -6,10 +6,10 @@
 from abr_control.utils import transformations
 
-from .path_planner import PathPlanner
+# from .path_planner import PathPlanner
```
delete

studywolf commented on README.rst:
```diff
@@ -66,52 +66,107 @@ If you would like to use the Pygame API, from your anaconda environment run::
+API for controlling the real arm. For installation instructions, see the
+`ABR_Jaco2 <https://github.com/abr/abr_jaco2/>`_ repository.
+
 PyDMPs
```
the DMP trajectory planner is gone, right? we don't need to mention alternative ways here, that's fine. we should also remove pydmps from any mention in the setup files

studywolf replied:
(i'll do this)

studywolf commented on abr_control/controllers/path_planners/path_planner.py:
```diff
@@ -1,16 +1,459 @@
+                )
+
+            self.orientation_path = np.asarray(self.orientation_path)
+            # TODO should this be included? look at proper derivation here...
```
it's not clear to me here what's being asked. should the angular velocities be included?

p3jawors replied:
Sorry meant to remove that, I checked some other links and it looks like angular velocity is just the rate of change of orientation. 
Check out the first equation under `angular velocity`
https://courses.lumenlearning.com/physics/chapter/6-1-rotation-angle-and-angular-velocity/

studywolf replied:
should i remove both the TODO and the hyperlink?

studywolf commented on abr_control/controllers/path_planners/path_planner.py:
```diff
@@ -37,57 +480,258 @@ def convert_to_time(self, path, time_length):
-
-    def _plot(self, target_position):
-        """Plot the generated trajectory
+    def _plot(self, start_position, target_position, ang=None):
```
what is the `ang` parameter?

p3jawors replied:
Since you can optionally plan an orientation path, that parameter sets whether or not we're plotting orientaiton. However, most of the parameters used in this function are `self` variables set in `generate_path`. Since it's only internally called after running `generate_path`, I think it would make the most sense to remove all of the parameters from `_plot()` and just have `start_position` , `target_position` set as `self` variables. There is enough information in `generate_path` that we can get whether or not an orientation path exists and omit the `ang` parameter all together. I can make the changes if you push your latest, if you'd like.

studywolf replied:
nooooo haha definitely it's preferred to explicitly pass things in as opposed to just using self variables. i've made a bunch of changes to this function, i'll push and then we can look at what that would be like

p3jawors replied:
The reason I did it this way is because the plotting function was only intended to be run internally when `plot=True` in `generate_path`. It uses variables that are created in `generate_path` that aren't passed back to the user, so I set it up with `self ` variables. But hey, you're Dr  Mr Bossman 👍

studywolf commented:
Went through, looks good overall! A couple notes on comments: 1) want to make sure the ones in """ """ are always starting with capital letters and using punctuation at the end, it was a bit inconsistent throughout, 2) we have an 88 char limit, it looked like some were formatted for 60 chars, 3) want to avoid "we" or "our" throughout (sorry i think you picked that up from me, i've been trying to correct it in my own commits).

In the plotting function there was a lot of repeated code; i reduced it quite a bit with some for loops. I believe it's all still plotting the same things but if you could closely compare that would be great, it's possible i missed some things.

I think there's still just the comment for `ang`, if you could add that in then, then i can do the squashing and it'll be ready to merge!

p3jawors commented:
Looked over the edits and they all look good (sorry for the grammar :sweat_smile: ). I tested the plotting and it works fine. I removed the`ang` parameter as we can just check the shape of the path. I also removed a couple redundant scatter plot points in one of the figures. LGTM to merge when you're ready :+1:

studywolf commented:
LGTM!
